source/slang/check.cpp

```diff
@@ -1833,24 +1837,22 @@ namespace Slang
 
             // Make sure that by-name lookup is possible.
             buildMemberDictionary(typeDeclRef.getDecl());
-
-            Decl* firstMemberOfName = nullptr;
-            typeDeclRef.getDecl()->memberDictionary.TryGetValue(name, firstMemberOfName);
-
-            if (!firstMemberOfName)
+            auto lookupResult = lookUpLocal(getSession(), this, name, typeDeclRef);
```

Comment:
It is a nice idea to reuse the lookup logic here, since that would seemingly handle `extension` declarations for free.

The one gotcha I’d be worried about that lookup usually also lets us look “through” inheritance, so what is to stop this process from looking up a member of `IFoo` via inheritance while trying to satisfy a requirement from the same interface?